src/qt/rpcconsole.cpp

```diff
@@ -464,6 +446,101 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
     }
 }
 
+/**
+ * Small and fast parser supporting console command syntax, with limited functionality.
+ * Splits a command line string into a vector with command at position 0, and parameters after
+ *
+ * @param[in]    strCommand  Command line to parse
+ *
+ * @return a vector of strings with command and parameters
+ */
+
+std::vector<std::string> RPCExecutor::parseHelper(const std::string& strCommand)
+{
+    std::vector<std::string> vec = SplitString(strCommand, " (),\n");
+    auto should_remove = [](const std::string& str) { return str.empty(); };
+    vec.erase(std::remove_if(vec.begin(), vec.end(), should_remove), vec.end());
+    return vec;
+}
+
+/**
+ * Catches console-only command before a RPC call is executed
+ *
+ * @param[in]    command  Command line to execute
+ * @param[in]    wallet_model  Wallet model to use
+ * @return  true if command was handled by this method (even on errors), false otherwise
+ *
+ */
+
+bool RPCExecutor::executeConsoleOnlyCommand(const std::string& command, const WalletModel* wallet_model)
+{
+    std::string result;
+    std::string address;
+    std::string executableCommand;
+    std::vector<std::string> parsedCommand{parseHelper(command)};
+
+    // Catch the console-only generate command with 2 or less parameters before RPC call is executed .
+    if (!parsedCommand.empty() && parsedCommand[0] == "generate" && parsedCommand.size() <= 3) {
+        // Default number of blocks is 1 if missing
+        const std::string nblocks{parsedCommand.size() > 1 ? parsedCommand[1] : "1"};
+        // An empty maxtries parameters will be modified by generatetoaddress' default value
+        const std::string maxtries{parsedCommand.size() > 2 ? parsedCommand[2] : ""};
+
+        // Generate address
+        if (!RPCConsole::RPCExecuteCommandLine(m_node, address, "getnewaddress\n", nullptr, wallet_model)) {
+            Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate new address"));
+        } else {
+            executableCommand = "generatetoaddress " + nblocks + " \"" + address + "\" " + maxtries;
+            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "generatetoaddress " + nblocks + " " + address + " " + maxtries + "\n", nullptr, wallet_model)) {
+                    Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate blocks"));
```

Comment:
This line is indented one tab stop too much, and same for the two lines just below (after the `else`).

Reply:
Indentation was introduced after running the script suggested by @ jarolrod and are apparently ok according to clang format standard.

Reply:
That's strange, when I run `clang-format-diff`, it moves this left, where they should be:

<details>
<summary>click to expand</summary>

```
$ git show -U0 | ./contrib/devtools/clang-format-diff.py -p1 -i -v
Formatting src/qt/rpcconsole.cpp
$ git diff
diff --git a/src/qt/rpcconsole.cpp b/src/qt/rpcconsole.cpp
index 1408b289f..63820464b 100644
--- a/src/qt/rpcconsole.cpp
+++ b/src/qt/rpcconsole.cpp
@@ -419,12 +419,12 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
 
         // Attempt to execute console-only commands
         if (!RPCExecutor::executeConsoleOnlyCommand(executableCommand, wallet_model)) {
-                // Send to the RPC command parser if not console-only
-                if (!RPCConsole::RPCExecuteCommandLine(m_node, result, executableCommand, nullptr, wallet_model)) {
-                    Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Parse error: unbalanced ' or \""));
-                    return;
-                }
-                Q_EMIT reply(RPCConsole::CMD_REPLY, QString::fromStdString(result));
+            // Send to the RPC command parser if not console-only
+            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, executableCommand, nullptr, wallet_model)) {
+                Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Parse error: unbalanced ' or \""));
+                return;
+            }
+            Q_EMIT reply(RPCConsole::CMD_REPLY, QString::fromStdString(result));
         }
     }
     catch (UniValue& objError)
@@ -492,10 +492,10 @@ bool RPCExecutor::executeConsoleOnlyCommand(const std::string& command, const Wa
             Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate new address"));
         } else {
             if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "generatetoaddress " + nblocks + " " + address + " " + maxtries + "\n", /*pstrFilteredOut=*/nullptr, wallet_model)) {
-                    Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate blocks"));
+                Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate blocks"));
             } else {
-                    std::string answer = "{\n  \"address\": \"" + address + "\",\n  \"blocks\": " + result + "\n}";
-                    Q_EMIT reply(RPCConsole::CMD_REPLY, QString::fromStdString("\n" + answer + "\n\n"));
+                std::string answer = "{\n  \"address\": \"" + address + "\",\n  \"blocks\": " + result + "\n}";
+                Q_EMIT reply(RPCConsole::CMD_REPLY, QString::fromStdString("\n" + answer + "\n\n"));
             }
         }
         return true;
```
</details>

Reply:
I'll try from my side and give it a go later this week.